core/src/main/java/org/apache/iceberg/MergingSnapshotProducer.java

```diff
@@ -682,7 +692,8 @@ private ManifestFile createManifest(int specId, List<ManifestFile> bin) throws I
       writer.close();
     }
 
-    ManifestFile manifest = writer.toManifestFile();
+    ManifestFile manifest = GenericManifestFile.copyOf(writer.toManifestFile())
```

Comment:
The new created manifest needs sequence number since the previously new added manifests maybe rewrite.